src/instructlab/config/init.py

```diff
@@ -72,61 +74,80 @@ def init(
 ):
     """Initializes environment for InstructLab"""
     ensure_storage_directories_exist()
-    clone_taxonomy_repo = True
-    if interactive:
-        if exists(DEFAULTS.CONFIG_FILE):
-            overwrite = click.confirm(
-                f"Found {DEFAULTS.CONFIG_FILE} in the current directory, do you still want to continue?"
-            )
-            if not overwrite:
-                return
-        click.echo(
-            "Welcome to InstructLab CLI. This guide will help you to setup your environment."
-        )
-        click.echo(
-            "Please provide the following values to initiate the "
-            "environment [press Enter for defaults]:"
-        )
-        taxonomy_path = utils.expand_path(
-            click.prompt("Path to taxonomy repo", default=taxonomy_path)
-        )
+    cfg = None
+    # If the caller used `ilab init` then the parent command is `ilab`, so we need to get the
+    # parameter source from the parent command. If the caller used `ilab config init` then the
+    if ctx.parent.command.name == "ilab":
+        param_source = ctx.parent.get_parameter_source("config_file")
+    # if the caller used `ilab config init` then the parent command is `config`, so we need to get
+    # the  parameter source from the parent of the parent command.
+    else:
+        param_source = ctx.parent.parent.get_parameter_source("config_file")
 
-    try:
-        taxonomy_contents = listdir(taxonomy_path)
-    except FileNotFoundError:
-        taxonomy_contents = []
-    if taxonomy_contents:
-        clone_taxonomy_repo = False
-    elif interactive:
-        clone_taxonomy_repo = click.confirm(
-            f"`{taxonomy_path}` seems to not exist or is empty. Should I clone {repository} for you?"
-        )
+    if param_source == click.core.ParameterSource.ENVIRONMENT:
+        model_path = ctx.obj.config.serve.model_path
+        taxonomy_path = ctx.obj.config.generate.taxonomy_path
+        taxonomy_base = ctx.obj.config.generate.taxonomy_base
+        cfg = ctx.obj.config
```

Comment:
I [think](https://github.com/instructlab/instructlab/pull/1847#discussion_r1690148608) this is never a `None`, so you could probably move `cfg = get_default_config()` from line 150 under the `else` branch for "default" (non-golden-env) case.